.github/workflows/go.yml

```diff
@@ -0,0 +1,129 @@
+name: build
+
+on:
+  push:
+    paths-ignore:
+      - 'api-tests/**'
+    branches:
+      - main
+      - release-*
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+  pull_request:
+
+jobs:
+  build:
+    name: Build
+
+    strategy:
+      fail-fast: false
+      matrix:
+        go:
+          - version: 1.16.x
+            may-fail: false
+          - version: tip
+            may-fail: true
+
+    continue-on-error: ${{ matrix.go.may-fail }}
+
+    runs-on: ubuntu-latest
+
+    env:
+      GOPATH: ${{ github.workspace }}
+      WORKING_DIR: ${{ github.workspace }}/src/github.com/percona/pmm
+
+    defaults:
+      run:
+        working-directory: ${{ env.WORKING_DIR }}
+
+    steps:
+      - name: Checkout code
+        uses: percona-platform/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: ${{ env.WORKING_DIR }}
+
+      - name: Set up Go release
+        if: matrix.go.version != 'tip'
+        uses: percona-platform/setup-go@v2
+        with:
+          go-version: ${{ matrix.go.version }}
+
+      - name: Set up Go tip
+        if: matrix.go.version == 'tip'
+        run: |
+          git clone --depth=1 https://go.googlesource.com/go $HOME/gotip
+          cd $HOME/gotip/src
+          ./make.bash
+          echo "GOROOT=$HOME/gotip" >> $GITHUB_ENV
+          echo "$HOME/gotip/bin" >> $GITHUB_PATH
+
+      - name: Enable Go modules cache
+        uses: percona-platform/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ matrix.os }}-go-${{ matrix.go-version }}-modules-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ matrix.os }}-go-${{ matrix.go-version }}-modules-
+
```

Comment:
I see that modules cached is not used for some reason and modules are downloaded each time.
Let's add here explicit step to resolve modules
```
      - name: Download Go modules
        run: go mod download
```